src/backend/distributed/utils/colocation_utils.c

```diff
@@ -1000,14 +1001,16 @@ ColocationGroupTableList(uint32 colocationId, uint32 count)
 	HeapTuple heapTuple = systable_getnext(scanDescriptor);
 	while (HeapTupleIsValid(heapTuple))
 	{
-		bool isNullArray[Natts_pg_dist_partition];
-		Datum datumArray[Natts_pg_dist_partition];
+		Datum* datumArray = (Datum *) palloc(tupleDescriptor->natts * sizeof(Datum));
```

Comment:
I'd allocate these outside & right before of the loop to avoid excessive amount of palloc calls.